pkg/scale/encode.go

```diff
@@ -62,6 +67,21 @@ type encodeState struct {
 }
 
 func (es *encodeState) marshal(in interface{}) (err error) {
+	marshalerType := reflect.TypeOf((*Marshaler)(nil)).Elem()
+	inv := reflect.ValueOf(in)
+	if inv.Type().Implements(marshalerType) {
+		methodVal := inv.MethodByName("MarshalSCALE")
+		values := methodVal.Call(nil)
+		if !values[1].IsNil() {
+			errIn := values[1].Interface()
+			err := errIn.(error)
+			return err
+		}
+		bytes := values[0].Interface().([]byte)
+		_, err = es.Write(bytes)
+		return
+	}
```

Comment:
```suggestion
        _, ok := in.(Marshaler)
        if ok {
            inv := reflect.ValueOf(in)
            methodVal := inv.MethodByName("MarshalSCALE")
	    values := methodVal.Call(nil)
	    if !values[1].IsNil() {
		  errIn := values[1].Interface()
		  err := errIn.(error)
		  return err
	    }
	    bytes := values[0].Interface().([]byte)
	    _, err = es.Write(bytes)
	    return
        }
```

Reply:
Ended up not using reflect at all to call `Marshaler.MarshalSCALE`.